src/LiveComponent/src/EventListener/LiveUrlSubscriber.php

```diff
@@ -68,34 +65,67 @@ public static function getSubscribedEvents(): array
     }
 
     /**
-     * @return array{
-     *     path: array<string, mixed>,
-     *     query: array<string, mixed>
-     * }
+     * @return array{ array<string, mixed>, array<string, mixed> }
      */
-    private function getLiveProps(MountedComponent $mounted): array
+    private function extractUrlLiveProps(MountedComponent $mounted): array
     {
-        $metadata = $this->metadataFactory->getMetadata($mounted->getName());
+        $pathProps = $queryProps = [];
 
-        $dehydratedProps = $this->liveComponentHydrator->dehydrate(
-            $mounted->getComponent(),
-            $mounted->getAttributes(),
-            $metadata
-        );
+        $mountedMetadata = $this->metadataFactory->getMetadata($mounted->getName());
 
-        $values = $dehydratedProps->getProps();
+        $dehydratedProps = $this->liveComponentHydrator->dehydrate($mounted->getComponent(), $mounted->getAttributes(), $mountedMetadata);
+        $props = $dehydratedProps->getProps();
```

Comment:
Unless I'm missing something, we still need to get dehydrated props from the mounted component, see https://github.com/symfony/ux/pull/2988

Reply:
Ah sure, it could be optimized in that way, thanks!